### PR TITLE
Added security checks

### PR DESCRIPTION
**Pull request by jhellar**
## Motivation

JIRA: https://issues.jboss.org/browse/AEROGEAR-2538, https://issues.jboss.org/browse/AEROGEAR-2539

## Progress Android

- [x] emulator check
- [x] developer mode check
- [x] screen lock check
- [x] backup check
- [x] encryption check
- [x] debugger check
- [x] root check

## Progress iOS

- [x] jailbreak check
- [x] emulator check
- [x] debug mode check
- [x] device lock check

### CODE REVIEW

**Comment by secondsun** [comment]
Looking

File: Security.Platform.Android/Checks/BackupDisallowedCheck.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+using Android.Content.PM;
+
+namespace AeroGear.Mobile.Security.Checks
+{
+    /// <summary>
+    /// Check to determine whether the allowBackup flag is enabled for the application.
+    /// </summary>
+    public class BackupDisallowedCheck : AbstractSecurityCheck
+    {
+        protected override string Name { get { return "Backup Flag Check"; } }
```
Comment (wojta) [inline Security.Platform.Android/Checks/BackupDisallowedCheck.cs]: Does `Name => "Backup Flag Check"` work here?

**Comment by wojta** [comment]
@secondsun is right about the directory structure

**Comment by ziccardi** [comment]
@jhellar @secondsun @wojta Let's make another PR after we have merged this and #47 #51 to move the projects

File: Security.Platform.iOS/Checks/NoDebuggerCheck.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    {
+        protected override string Name { get { return "Debugger Check"; } }
+
+        public NoDebuggerCheck()
+        {
+        }
+
+        public override SecurityCheckResult Check()
+        {
+#if DEBUG
```
Comment (ziccardi) [inline Security.Platform.iOS/Checks/NoDebuggerCheck.cs]: I think this should be removed

Comment (jhellar) [reply Security.Platform.iOS/Checks/NoDebuggerCheck.cs]: That is the actual check. Same approach is used in ios-sdk - https://github.com/aerogear/aerogear-ios-sdk/blob/master/modules/security/checks/NonDebugCheck.swift#L21-L25

**Comment by jhellar** [comment]
Thanks for review. All issues should be addressed now.

**Comment by secondsun** [comment]
@jhellar I'm taking one more look and will let you know :)

File: Security.Platform.Android/Checks/ScreenLockCheck.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+        public ScreenLockCheck(Context ctx)
+        {
+            this.context = ctx;
+        }
+
+        public override SecurityCheckResult Check()
+        {
+            KeyguardManager manager = (KeyguardManager)context.GetSystemService(Context.KeyguardService);
            return new SecurityCheckResult(this, manager.IsDeviceSecure);
```
Comment (ziccardi) [inline Security.Platform.Android/Checks/ScreenLockCheck.cs]: Reformat this code

**Comment by secondsun** [comment]
@jhellar Can you rebase on master?

**Comment by jhellar** [comment]
@secondsun rebased.

**Comment by jhellar** [comment]
@ziccardi fixed the issue with newline symbol.

File: Security.Platform.Android/Checks/DeveloperModeDisabledCheck.cs [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        private readonly Context context;
+
+        public DeveloperModeDisabledCheck(Context ctx)
+        {
+            this.context = ctx;
+        }
+
+        public override SecurityCheckResult Check()
+        {
+            bool devModeEnabled = Settings.Secure.GetInt(
```
Comment (aidenkeating) [inline Security.Platform.Android/Checks/DeveloperModeDisabledCheck.cs]: When running this in an emulator I'm seeing a `SettingNotFoundException` being thrown.

```
[AndroidRuntime] android.runtime.JavaProxyThrowable: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> Android.Provider.Settings+SettingNotFoundException: development_settings_enabled
[AndroidRuntime]   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <f32579baafc1404fa37ba3ec1abdc0bd>:0 
[AndroidRuntime]   at Java.Interop.JniEnvironment+StaticMethods.CallStaticIntMethod (Java.Interop.JniObjectReference type, Java.Interop.JniMethodInfo method, Java.Interop.JniArgumentValue* args) [0x00069] in <09bf3e262b934ffab2ba01f9fc7fd54d>:0 
[AndroidRuntime]   at Java.Interop.JniPeerMembers+JniStaticMethods.InvokeInt32Method (System.String encodedMember, Java.Interop.JniArgumentValue* parameters) [0x00018] in <09bf3e262b934ffab2ba01f9fc7fd54d>:0 
[AndroidRuntime]   at Android.Provider.Settings+Secure.GetInt (Android.Content.ContentResolver cr, System.String name) [0x0004b] in <25661073a35344a89f215a4cf81af37c>:0 
[AndroidRuntime]   at AeroGear.Mobile.Security.Checks.DeveloperModeDisabledCheck.Check () [0x0000c] in <3af04addddc1407fafe20c72004681b5>:0 
[AndroidRuntime]   at AeroGear.Mobile.Security.Executors.Sync.SyncSecurityCheckExecutor.Execute () [0x0001f] in <0e8c095b6cac4364a4d9747999250031>:0 
```

Comment (jhellar) [reply Security.Platform.Android/Checks/DeveloperModeDisabledCheck.cs]: Should be fixed now.

**Comment by secondsun** [comment]
ScreenLockCheck doesn't work when I tested it.

**Comment by secondsun** [comment]
@jhellar I've included a fix in a future PR, merging.

**Pull request merged**